### ISSUE TITLE
counter culture on has_many relationship not updating on destroy

### ISSUE THREAD

**Issue by shereenqumsieh** [magnusvk/counter_culture#265]
Hi there, love this gem, but I'm running into an issue with my has_many relationship that I can't quite figure out. Any help is really appreciated. I have the following:

Idea.rb
```
has_many :idea_tags, dependent: :destroy
has_many :tags, through: :idea_tags
```

Tag.rb
```
has_many :idea_tags, dependent:  :destroy
has_many :ideas, through: :idea_tags
```

IdeaTag.rb
```
belongs_to :idea
belongs_to :tag
counter_culture :tag, column_name: "ideas_count"
```

I find the above works great when adding a tag to an idea. It will update the counter on the Tag model, so @tag.ideas_count increases as it should. However, I find if i remove the tag from the idea, it doesn't decrement the count.

Am I doing something wrong? Should this work as expected? Should the count decrease?

**Comment by shereenqumsieh** [comment]
If it's worth noting, that when I run `IdeaTag.counter_culture_fix_counts` it updates it correctly. I was hoping it could update automatically when a tag is removed from Idea.

**Comment by magnusvk** [comment]
Yes, the count should decrease. [There's a test for this](https://github.com/magnusvk/counter_culture/blob/master/spec/counter_culture_spec.rb#L87) so I'm confident this works in principle.

Are you deleting these without callbacks? counter_culture can only keep counts up-to-date if callbacks are run, so you have to call `.destroy`

**Comment by shereenqumsieh** [comment]
Thanks Magnus, I think I see what you mean. I was simply doing this to create/update the tags for an Idea:

`params[:idea].permit(:title, :summary, :creator, :priority, tag_ids: [])`

What you're saying is, I shouldn't do it this way, but instead I have to call a destroy on the IdeaTag model? Appreciate the response!

**Comment by magnusvk** [comment]
Hm—that's weird, I don't immediately know why this doesn't work. Perhaps assigning `tag_ids` directly destroys the `IdeaTag`s without callbacks? I understand why you'd expect this to work, but at the end of the day if Rails decides to do this without callbacks, there's really nothing counter_culture can do to maintain counts. Why don't you try manually managing the `IdeaTag`s here—that could be a good workaround for you.

**Comment by shereenqumsieh** [comment]
Hi Magnus, You are right in that the **after_destory** callback was not being triggered. And definitely, it was a Rails issue not counter_culture, so I appreciate you taking the time to respond - you helped point me in the right direction and find a solution.

I thought I'd update in case anyone else has this issue. Reading through a ton of rails stuff, it seems you have to add the dependent: :destroy to the through association to trigger the callbacks properly. Here is a good summary of the issue: https://github.com/rails/rails/issues/29078. It appears to have been fixed way back in 3.1: https://github.com/rails/rails/issues/1962.

So changing to this helped me ensure that the **after_destroy** callback was triggered:

`has_many :idea_paths
  has_many :paths, through: :idea_paths, dependent: :destroy`

Thanks for your help!
